src/components/NumberBadge/index.tsx

```diff
@@ -6,43 +6,42 @@ export interface NumberBadgeProps {
 }
 
 export const NumberBadge = styled.div<NumberBadgeProps>`
-  position: relative;
-  width: 16px;
-  height: 16px;
-  border: 1px solid ${props => props.theme.bg3};
-  border-radius: 50%;
+ 
+  display:flex;
+  padding:0 2px;
+  height:16px;
+  width: auto;
+  border: 1.75px solid ${props => props.theme.bg3};
+  border-radius: 4px;
   background-color: ${props => transparentize(0.3, props.theme.bg3)};
 
   & > span {
-    position: absolute;
-    top: 50%;
-    left: 50%;
-    transform: translate(-50%, -50%);
+  
     font-weight: 500;
-    font-size: 11px;
+    font-size: 9px;
   }
   
   ${({ badgeTheme: theme }) =>
     theme === 'green' &&
     css`
       background: rgba(14, 159, 110, 0.08);
-      border: 1px solid rgba(14, 159, 110, 0.65);
+      border: 1.75px solid rgba(14, 159, 110, 0.65);
       color: #0e9f6e;
     `}
   
   ${({ badgeTheme }) =>
     badgeTheme === 'orange' &&
     css`
       background: rgba(242, 153, 74, 0.08);
-      border: 1px solid rgba(242, 153, 74, 0.65);
+      border: 1.75px solid rgba(242, 153, 74, 0.65);
       color: #f2994a;
     `}
 
   ${({ badgeTheme }) =>
     badgeTheme === 'red' &&
     css`
       background: rgba(240, 46, 81, 0.08);
-      border: 1px solid rgba(240, 46, 81, 0.65);
+      border: 1.75px solid rgba(240, 46, 81, 0.65);
```

Comment:
The border properties can be broken down and DRY up the code. Alternatively, a hash table can be used to map the colors to their respective CSS properties.